### ISSUE TITLE
Cyrillic glyphs do not appear in the built font file

### ISSUE THREAD

**Issue by Verororo** [fontforge/fontforge#4616]
_Windows 10, FontForge version 20201107_

Hello. There is one bitmap font I really like, though it doesn't have cyrillic, so I wanted to use FontForge to edit it, draw the cyrillic glyphs myself and be able to write in it.
I opened the needed font in FontForge, switched the encoding to Unicode, found the glyphs I need and drew them. They appeared with a gray X in the background, like that. 
![graycross](https://user-images.githubusercontent.com/39249949/107119098-0137a200-688e-11eb-92e6-489e182efb30.PNG)
I fixed it by copying the width of an existing glyph and pasting it into the marked ones. The Xes disappeared.
Then I built it in the Win FON bitmap format. When I opened the .fon file, Windows font viewing program showed the cyrillic letters as the default ones, not the ones I drew.
![fontviewing](https://user-images.githubusercontent.com/39249949/107119308-6b9d1200-688f-11eb-9341-4d1ddc977414.PNG)
When choosing the installed font in any text editor program, the cyrillic letters don't appear. Instead, the blank spaces are showed, as if the new glyphs were not added. They don't appear when opening the built file in FontForge, too. The encoding in the built file is Win ANSI, not Unicode.
What I was expecting was that the encoding in the built file would be Unicode and would include the cyrillic glyphs I tried to add. And in reality, nothing of that happens.

So, I'm guessing that the problem is with the encoding. Though in the FontForge I re-encoded the font to Unicode, the built file is being built with ANSI encoding, which does not have cyrillic. But that is just a guess. If it is right, how else would I change it if the Encoding menu doesn't do it?

[(Link to .sfd file)](https://drive.google.com/file/d/1A8T_i60s_bLGF5tZPE8vtmxNJFzjp0Ae/view?usp=sharing)

### When reporting a bug/issue:
- [x] Screenshot
- [x] The FontForge version and the operating system you're using
- [x] The behavior you expect to see, and the actual behavior
- [x] Steps to reproduce the behavior 
- [ ] \(optional) Possible solution/fix/workaround

**Comment by skef** [comment]
Looking at the source code I don't think FontForge is going to be able to write your file. It always writes `0` into the `dfCharSet` header field, which appears to correspond to Windows ANSI and that doesn't include a Cyrillic encoding.

Based on https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-wmf/0d0b32ac-a836-4bd2-a112-b6000a1b4fc9 and https://docs.microsoft.com/en-us/previous-versions/aa915041(v=msdn.10)?redirectedfrom=MSDN it looks like the encoding you want is probably "Microsoft code page 1251". Happily that's in [Encodings.ps](https://fontforge.org/downloads/Encodings.ps.gz) pointed to by [this](https://fontforge.org/docs/ui/menus/encodingmenu.html#user-defined-encodings) section of the FontForge documentation. If you download that file, `gunzip` it, and then add it by choosing "Encoding -\> Load Encoding..." from the menu, CodePage1251 will be available. You can then "Encoding -\> Reencode" with it and arrange your glyphs appropriately. 

After that, though, you'll still have the wrong `dfCharSet` entry in the generated font. There are two options here. One is to binary-edit the resulting file and change the field to 0xCC (204 in decimal). The file format is described [here](https://web.archive.org/web/20100130073457/http://support.microsoft.com/kb/65123). The other option is to recompile FontForge with (I think) `putc((char)0xCC, file);` on line 554 of `fontforge/winfonts.c` (and only use that binary for this purpose).

However, no guarantees about any of this. You might have better luck with the [PSF Tools](https://www.seasip.info/Unix/PSF/). 

(Unfortunately given that you're on Windows much of this advice may be hard to follow.)

**Comment by Verororo** [comment]
Thank you for the reply. Recompiling it seems pretty hard because I need to download the libraries and I have no experience with CMake. So I was trying to binary edit the file, but couldn't find the address where `dfCharSet` would be...

**Comment by skef** [comment]
Can you post a screenshot (or something) of the first bytes in the generated file? This could be a FNT/FON problem -- that era of windows formats is confusing.

**Comment by skef** [comment]
I mean a screenshot of the hex values, I suppose.

**Comment by skef** [comment]
(Or, alternatively, you might have luck with [fony](http://hukka.ncn.fi/?fony). It's oldish but it probably still runs. You could construct the whole font that way or try saving with the 1251 encoding like I suggested and just change the charset.)

**Comment by Verororo** [comment]
Sure, here's the screenshot.
![hexvalues](https://user-images.githubusercontent.com/39249949/107158121-a175ef80-6990-11eb-8649-d65374b246d3.PNG)
That's only the part of it. I can upload the file so you can see all the values.

**Comment by skef** [comment]
The line starting at offset 190 should read:

```
00000190: 0000 0000 0000 0090 01cc 0a00 1400 110a  ................
```

Now, that's just the charset variable -- no guarantee this will work in the larger picture.

**Comment by Verororo** [comment]
Gah, this didn't help.

Actually, I just found the version of this font with cyrillic after a long search, so this is no longer a problem for me. Thanks for your feedback.
